lib/role.rb

```diff
@@ -0,0 +1,38 @@
+class Role 
+  attr_accessor :role_type, :proposal, :user
+  def initialize(user,proposal)
+    self.user = user
+    self.proposal = proposal
+    self.set_role_type
+  end
+
+  def set_role_type
+    role = ''
+    if self.user == self.proposal.requester
+      role = 'requester'
+    elsif self.proposal.is_active_approver?(self.user)
+      role = 'active approver'
+    elsif self.proposal.observers.include?(self.user)
+      role = 'observer'
+    elsif self.proposal.approvers.include?(self.user)
+      role = 'approver'
+    end
+    self.role_type = role
```

Comment:
Probably want to use symbols rather than strings.

More importantly, can't a single user play multiple roles on a proposal? If so, maybe you want to store multiple roles and encourage a capabilities model. Right now, we're asking "Are you an approver?" Maybe we should ask, instead "Can you approve?" "Are you subscribed?". This gets close to Pundit, so I'm not quite sure how to separate the two.

Be sure to reach out to hash this through with @annalee who's been thinking about this more.